tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -57,38 +57,57 @@ func shouldUseBinaryForCurrentContext(iptablesBin string) (IptablesVersion, erro
 	// if we find one unless the host OS is badly broken we will find the others.
 	iptablesSaveBin := fmt.Sprintf("%s-save", iptablesBin)
 	iptablesRestoreBin := fmt.Sprintf("%s-restore", iptablesBin)
+	var parsedVer *utilversion.Version
+	var isNft bool
 	// does the "xx-save" binary exist?
+	_, binExistsErr := exec.Command(iptablesSaveBin).CombinedOutput()
```

Comment:
running `iptables-save` actually does some operations. Do we really need to run it?

I think we can just do `exec.LookPath` and then `os.Stat` to see if the file exists?

Reply:
We can, but we gotta run it right after anyway to detect if any rules are present, so I'm not inclined to think it will save much.